### PR TITLE
Customizable error color

### PR DESCRIPTION
**Pull request by strouhaljan**
JIRA: ONE-4648

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1522/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3098/ : FAILURE in 10m 43s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1523/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3100/ : SUCCESS in 11m 01s

**Comment by strouhaljan** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1529/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3105/ : SUCCESS in 10m 53s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1531/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/494/ : SUCCESS in 18m 45s

File: libs/sdk-ui-kit/styles/scss/variables.scss [hunk 1]
```diff
@@ -40,6 +40,11 @@ $gd-palette-primary-dimmed: var(
...
 $gd-palette-primary-base-darken06: var(--gd-palette-primary-base-d06, darken($gd-color-highlight, 6%));
 
+// error color
+$gd-palette-error-base: var(--gd-palette-error-base, $gd-color-negative);
+// error derived colors
+$gd-palette-error-dimmed: var(--gd-palette-error-dimmed, mix($gd-color-negative, $gd-color-text-light, 10%));
+
 // secondary colors
 $gd-color-positive-dimmed: mix($gd-color-positive, $gd-color-text-light, 10%);
 $gd-color-negative-dimmed: mix($gd-color-negative, $gd-color-text-light, 10%);
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/styles/scss/variables.scss:50]: this should be removed and replaced by variable from line 46

File: libs/sdk-ui-theme-provider/src/cssProperties.ts [hunk 2]
```diff
@@ -102,71 +102,151 @@ export function parseThemeToCssProperties(
...
+        "palette-primary-dimmed50",
+        palette?.primary?.base && mix(0.5, palette.primary.base, GD_COLOR_TEXT_LIGHT),
+    ),
+    getCssProperty(
+        "palette-primary-lightest",
+        palette?.primary?.base && setLightness(0.96, palette.primary.base),
+    ),
+    getCssProperty("palette-error-lightest", palette?.error?.base && setLightness(0.96, palette.error.base)),
+    getCssProperty(
+        "palette-error-base-mix15-white",
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-theme-provider/src/cssProperties.ts:126]: dont you want to keep naming same as for `palette-primary-dimmed50`?

**Review by ivan-nejezchleb — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1535/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3111/ : SUCCESS in 10m 45s

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/498/ : SUCCESS in 23s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/758/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3112/ : SUCCESS in 10m 37s

**Pull request merged**